aws/rust-runtime/aws-types/src/os_shim_internal.rs

```diff
@@ -208,11 +214,81 @@ mod env {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct TimeSource(time_source::Inner);
+
+impl TimeSource {
+    pub fn real() -> Self {
+        TimeSource(time_source::Inner::Real)
+    }
+
+    pub fn manual(time_source: &ManualTimeSource) -> Self {
+        TimeSource(time_source::Inner::Manual(time_source.clone()))
+    }
+
+    pub fn now(&self) -> SystemTime {
+        match &self.0 {
+            Inner::Real => SystemTime::now(),
+            Inner::Manual(manual) => manual.now(),
+        }
+    }
+}
+
+impl Default for TimeSource {
+    fn default() -> Self {
+        TimeSource::real()
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct ManualTimeSource {
+    queries: Arc<Mutex<Vec<SystemTime>>>,
```

Comment:
Shouldn't there be a way to access `queries` to confirm it in tests?

Reply:
yeah, will add. I hadn't added it since it wasn't needed yet.